core/trino-main/src/main/java/io/trino/operator/output/PagePartitioner.java

```diff
@@ -444,11 +446,16 @@ private Page getPartitionFunctionArguments(Page page)
         return new Page(page.getPositionCount(), blocks);
     }
 
+    @Override
     public void close()
     {
-        flushPositionsAppenders(true);
-        flushPageBuilders(true);
-        memoryContext.close();
+        try {
+            flushPositionsAppenders(true);
+            flushPageBuilders(true);
+        }
+        finally {
+            memoryContext.close();
```

Comment:
You should reset builders memory also before you close memory context. Are you sure `PagePartitioner` will actually be released as an object itself?

Reply:
flush will reset the builders memory.
`PagePartitioner` is either freed on `PagePartitionerPool.release` or when `PartitionedOutputOperatorFactory` is freed along with `PagePartitionerPool`

Reply:
> flush will reset the builders memory.

Only if one of the builders in sequence doesn't fail. So here you:
1. flush builders.
2. Release memory `memoryContext.close()` even when flush failed and memory might be still allocated. You should call `memoryContext.close()` only after builders are released

Technically, you should call `io.trino.operator.output.PagePartitioner#updateMemoryUsage` if builders are not nullified

Reply:
ok, that makes sense. I added clearing the buffer arrays